packages/labs/analyzer/src/test/javascript/mixins_test.ts

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {suite} from 'uvu';
+// eslint-disable-next-line import/extensions
+import * as assert from 'uvu/assert';
+
+import {
+  AnalyzerTestContext,
+  languages,
+  setupAnalyzerForTest,
+} from '../utils.js';
+import {ClassDeclaration, LitElementDeclaration} from '../../lib/model.js';
+import {DiagnosticCode} from '../../lib/diagnostic-code.js';
+
+for (const lang of languages) {
+  const test = suite<AnalyzerTestContext>(`Mixin tests (${lang})`);
+
+  test.before((ctx) => {
+    setupAnalyzerForTest(ctx, lang, 'mixins');
+  });
+
+  test('basic mixin declaration', ({getModule}) => {
+    const decl = getModule('mixins').getDeclaration('Highlightable');
+    assert.ok(decl.isMixinDeclaration());
+    assert.instance(decl.classDeclaration, ClassDeclaration);
+    assert.instance(decl.classDeclaration, LitElementDeclaration);
+    const classDecl = decl.classDeclaration as LitElementDeclaration;
+    assert.equal(classDecl.name, 'HighlightableElement');
+    assert.ok(classDecl.reactiveProperties.get('highlight'));
+  });
+
+  test('basic mixin usage', ({getModule}) => {
+    const decl = getModule('element-a').getDeclaration('ElementA');
+    assert.ok(decl.isLitElementDeclaration());
+    assert.ok(decl.heritage.superClass);
+    assert.equal(decl.heritage.superClass.name, 'LitElement');
+    assert.equal(decl.heritage.mixins.length, 1);
+    assert.equal(decl.heritage.mixins[0].name, 'Highlightable');
+  });
+
+  test('complex mixin declaration A', ({getModule}) => {
+    const decl = getModule('mixins').getDeclaration('A');
+    assert.ok(decl.isMixinDeclaration());
+    assert.instance(decl.classDeclaration, ClassDeclaration);
+    assert.instance(decl.classDeclaration, LitElementDeclaration);
+    const classDecl = decl.classDeclaration as LitElementDeclaration;
+    assert.equal(classDecl.name, 'A');
+    if (lang === 'ts') {
+      //TODO(kschaaf): Add detection of constructor-defined fields to ClassField
```

Comment:
there was some preliminary work on this in #4135, so you may be able to remove the condition here